Update docs related to supported python version [ENG 199] (#12109)
RasaHQ/rasa: docs/docs/installation/environment-set-up.mdx

:::note Python3 Supported Versions

> Currently, rasa supports the following Python versions: `3.7`, `3.8`, `3.9` and `3.10`.

Reviewer: We also need to mention that `3.10` is only available from `3.4.x` onwards.

Note that Python `3.10` is not currently supported for rasa installations on Apple Silicon.
The blocker consists of current inability to update to Tensorflow version >= `2.10.x` which is pending the resolution

> of a Tensorflow bug.

Reviewer: We need to specify here that  Python `3.10` on Apple Silicon doesn't work in `3.4.x`, but will be supported from `3.5.x` onwards.

:::note Python3 Supported Versions

Currently, rasa supports the following Python versions: `3.7`, `3.8`, `3.9` and `3.10`.

> It's worth noting that Python `3.10` is only available for versions `3.4.x` onwards.

Reviewer: ```suggestion
Note that Python `3.10` is only supported for versions `3.4.x` and upwards.
```
Hopefully `upwards` is not considered a typo also 😅

> Although, on Apple Silicon, Python `3.10` isn't functional in `3.4.x` but will be supported starting from `3.5.x`.

Reviewer: ```suggestion
Additionally, rasa installation on Apple Silicon with Python `3.10` is not functional in `3.4.x` but will be supported starting from `3.5.x`.
```
